usr/share/rear/skel/default/etc/scripts/unlock-opal-disks

```diff
@@ -201,6 +201,20 @@ while (( attempt < max_authentications )); do
         else
             display_message "$unlocked_device_count of $device_count disks unlocked, rebooting..."
         fi
+        if type -p efibootmgr &>/dev/null; then
+            # Workaround for EFI firmware glitch which prevents booting the real OS after unlocking SEDs.
+            # Symptom: Boot hangs when Grub hands over control to the Linux kernel. Seen on an HPE ML10Gen9 server.
+            # Cf. https://github.com/rear/rear/issues/2425
+            read -r field_name current_boot_number field_junk <<<"$(efibootmgr 2>/dev/null | grep 'BootCurrent:')"
+            if [[ "$current_boot_number" =~ ^[0-9]+$ ]]; then
+                # Set the current boot number as the boot entry for the next boot. This does not make much sense,
+                # except that we can be sure that we're setting a valid boot entry which we can then remove.
+                efibootmgr --bootnext "$current_boot_number" --quiet
+                # Remove the next boot entry. This should convince the firmware to use the boot order to determine
+                # the effective boot entry for the next boot.
+                efibootmgr --delete-bootnext --quiet
+            fi
```

Comment:
When $current_boot_number has not the expected form
would it make sense to `echo` some appropriate message to the user
to have him at least informed that things could go wrong here?

Reply:
I wanted the code to be completely silent even in case of errors as it is normally not required to work at all. So if `efibootmgr` spits out something unexpected, silent failure would not disturb or delay the boot process. This fix deals with a probably very rare condition and should not compromise a pleasant boot experience for others.

However, I just noticed that boot entry numbers can contain hex digits. If you happen to see more than 10 boot entries, and boot from entry `a` or above, the code would fail. I'll post another commit to account for that improbable case, too.

Reply:
FYI: It seems
https://github.com/rear/rear/issues/2276#issuecomment-559879836
shows an interesting example of boot entry numbers with hex digits.

Reply:
Yes, that one looks interesting.

If entries have piled up from testing, I usually use `efibootmgr` to clean up the mess. Not each system setup allows a user to really delete boot entries.

If a large number of boot entries is actually in use, I hope that they have excellent documentation - or better automation - in place to stay in control. ;-)